src/nyc_trees/apps/users/views/group.py

```diff
@@ -227,10 +231,34 @@ def request_mapper_status(request):
 
 
 def group_unmapped_territory_geojson(request, group_id):
-    blockfaces = Blockface.objects.filter(
-        is_available=True,
-        territory__group_id=group_id
-    )
+    # Get unmapped blockfaces
+    blockfaces = Blockface.objects.filter(is_available=True)
+
+    my_territory_q = Q(territory__group_id=group_id)
+
+    if request.body:
+        # Get potentially selectable blockfaces in polygon
+        # (those in my territory or unclaimed)
+        point_list = json.loads(request.body)
+        point_list.append(point_list[0])  # Close the polygon
+        polygon = Polygon((point_list))
+
+        no_reservations_q = \
+            Q(blockfacereservation__isnull=True) \
+            | Q(blockfacereservation__canceled_at__isnull=False) \
+            | Q(blockfacereservation__expires_at__lt=now())
+        nobodys_territory_q = Q(territory__group_id=None)
+        unclaimed_q = no_reservations_q & nobodys_territory_q
+
+        blockfaces = blockfaces \
+            .filter(geom__within=polygon) \
```

Comment:
We can see what others think. I want to draw a corral around the blocks I want and not worry about the ones I intersect.